app/controllers/student_labs_controller.rb

```diff
@@ -3,6 +3,10 @@ class StudentLabsController < ApplicationController
 
   def index
     @student_labs = current_user.student_labs
+    @ignorables = []
+    @student_labs.each do |student_lab|
+      @ignorables << student_lab.lab_id if student_lab.complete
```

Comment:
What if the student already has a incomplete student_lab record, then the student is able to add another student_lab record that point to the same lab but marked as complete instead of update the existing one to complete status.